backend/config/corpora-api.yml

```diff
@@ -604,6 +604,54 @@ paths:
         "403":
           $ref: "#/components/responses/403"
 
+  /dp/v1/datasets/index:
+    get:
+      tags:
+        - datasets
+      summary: List datasets for indexing purposes
+      security:
+        - cxguserCookieLenient: []
+        - {}
+      description: >-
+        STUB
+      operationId: backend.corpora.lambdas.api.v1.dataset.get_datasets_index
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  type: object
+                  properties:
+                    id:
+                      type: string
+                    name:
+                      type: string
+                    collection_id:
```

Comment:
Will this be a filterable field in the UI? Since it's an ID, guessing not. And if not, is it needed?

Reply:
This is required because the frontend will have to join each dataset with its collection to retrieve (at minimum) the collection name.

Reply:
We discussed this with @MillenniumFalconMechanic and @NoopDog and we decided for this approach. I think the rationale is that a list of collections is needed regardless for the other view. Adding the collection_name here isn't free because it needs another query, so I'd leave it out for now. We can add it back later if required.